rest/data/project_settings.go

```diff
@@ -470,13 +470,22 @@ func SaveProjectSettingsForSection(ctx context.Context, projectId string, change
 		if err = parsley.ValidateFilters(mergedSection.ParsleyFilters); err != nil {
 			return nil, errors.Wrap(err, "invalid Parsley filters")
 		}
+	// This section does not support repo-level at this time.
 	case model.ProjectPageGithubAppSettingsSection:
+		mergedSection.Id = mergedBeforeRef.Id
+		if changes.GithubAppAuth.AppID != int(before.GitHubAppAuth.AppID) {
+			if err = mergedSection.SetGithubAppCredentials(int64(changes.GithubAppAuth.AppID), []byte(utility.FromStringPtr(changes.GithubAppAuth.PrivateKey))); err != nil {
```

Comment:
Kinda smart since this ensures that if users set GitHub app auth, they have to set both app ID and secret to overwrite either of them. 😎 

Just had some difficulty reviewing a PR where we had to painfully un-redact a secret to ensure it didn't get overwritten by the redacted placeholder field.

Reply:
The UI will make it so that users have to fully remove all app credentials (GitHub ID & private key) if they want to change their app credentials configuration! There will be a button that says something along the lines of `"Delete credentials"`